dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -69,12 +73,13 @@
       = f.select :user_type, user_type_options
 
   %div= f.submit t('crud.update'), class: 'btn btn-default'
-%h3= t('user.delete')
-%p
-  = t('user.delete_text')
-  - if current_user.teacher?
-    = t('user.delete_text_teacher_note')
-  = button_to t('user.delete'), registration_path(resource_name), data: { confirm: t('user.delete_confirm') }, method: :delete, class: 'btn btn-danger'
+- if current_user.can_delete_account?
```

Comment:
Throughout, the HAML has been using `f.object` rather than `current_user`. Why the change? Should the rest of the HAML be using `current_user`? Should this be using `f.object`?